code/espurna/system.ino

```diff
@@ -97,19 +110,21 @@ void systemLoop() {
     // Heartbeat
     // -------------------------------------------------------------------------
 
-    #if HEARTBEAT_MODE == HEARTBEAT_ONCE
-        if (_system_send_heartbeat) {
-            _system_send_heartbeat = false;
-            heartbeat();
-        }
-    #elif HEARTBEAT_MODE == HEARTBEAT_REPEAT
+    if (_system_send_heartbeat && _heartbeat_mode == HEARTBEAT_ONCE) {
+        heartbeat();
+        _system_send_heartbeat = false;
+    } else if (_heartbeat_mode == HEARTBEAT_REPEAT || _heartbeat_mode == HEARTBEAT_REPEAT_STATUS) {
         static unsigned long last_hbeat = 0;
-        if (_system_send_heartbeat || (last_hbeat == 0) || (millis() - last_hbeat > HEARTBEAT_INTERVAL)) {
-            _system_send_heartbeat = false;
+        #if NTP_SUPPORT
+            if ((_system_send_heartbeat && ntpSynced()) || (millis() - last_hbeat > _heartbeat_interval)) {
```

Comment:
Why checking for `ntpSynced` here?

Reply:
I have added ntpSynced because if we have report interval ONCE, I have noticed that on first report sometimes datatime value is not published because of not synced ntp. I was thinking that it would be nice to get the datatime on device startup to check if everything is ok.